server/app.py

```
@@ -55,6 +63,35 @@ async def lifespan(app: FastAPI):
+    if has_subject_index:
+        pending_registered = False
+        for attempt in range(3):
+            try:
+                await register_to_backend_pending(
+                    public_url,
+                    settings.dual_2pc_subject_index,
+                    settings.engine_secret_key,
+                )
+                pending_registered = True
+                break
+            except Exception as e:  # 분기 2 내부 catch — 외부 SystemExit 차단함
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)